packages/blockchain-wallet-v4-frontend/src/modals/DebitCard/OrderCard/ResidentialAddress/ResidentialAddress.tsx

```diff
@@ -0,0 +1,361 @@
+import React, { useEffect } from 'react'
+import { FormattedMessage } from 'react-intl'
+import { useDispatch, useSelector } from 'react-redux'
+import { validate } from 'postal-codes-js'
+// @ts-ignore
+import postalCodes from 'postal-codes-js/generated/postal-codes-alpha2'
+import { path } from 'ramda'
+import { Field, InjectedFormProps, reduxForm } from 'redux-form'
+import styled from 'styled-components'
+
+import { CountryScope } from '@core/types'
+import { BlockchainLoader, Button, HeartbeatLoader, Text } from 'blockchain-info-components'
+import FlyoutContainer from 'components/Flyout/Container'
+import FlyoutContent from 'components/Flyout/Content'
+import FlyoutFooter from 'components/Flyout/Footer'
+import FlyoutHeader from 'components/Flyout/Header'
+import Form from 'components/Form/Form'
+import FormGroup from 'components/Form/FormGroup'
+import FormItem from 'components/Form/FormItem'
+import FormLabel from 'components/Form/FormLabel'
+import SelectBox from 'components/Form/SelectBox'
+import TextBox from 'components/Form/TextBox'
+import { actions, model, selectors } from 'data'
+import { ResidentialAddress as ResidentialAddressFormValues } from 'data/components/debitCard/types'
+import { RootState } from 'data/rootReducer'
+import { CountryType, StateType } from 'data/types'
+import { useCountryList, useRemote, useUSStateList } from 'hooks'
+import { countryUsesZipcode, required } from 'services/forms'
+
+const { RESIDENTIAL_ADDRESS_FORM } = model.components.debitCard
+
+const countryUsesPostalCode = (countryCode) => {
+  return path([countryCode, 'postalCodeFormat'], postalCodes)
+}
+
+const requiredZipCode = (value, allVals) => {
+  const countryCode = (path(['country', 'code'], allVals) || path(['country'], allVals)) as string
+  if (!path([countryCode, 'postalCodeFormat'], postalCodes)) return undefined
+  if (!value)
+    return (
+      <div data-e2e='requiredMessage'>
+        <FormattedMessage id='formhelper.required' defaultMessage='Required' />
+      </div>
+    )
+
+  return validate(countryCode, value) === true ? undefined : (
+    <FormattedMessage id='formhelper.requiredzipcode' defaultMessage='Invalid zipcode' />
+  )
+}
+
+const getCountryElements = (countries: Array<CountryType>) => [
+  {
+    group: '',
+    items: countries.map((country: CountryType) => ({
+      text: country.name,
+      value: country.code
+    }))
+  }
+]
+
+const getStateElements = (states: Array<StateType>) => [
+  {
+    group: '',
+    items: states.map((state: StateType) => ({
+      text: state.name,
+      value: state.code
+    }))
+  }
+]
+
+const CustomForm = styled(Form)`
```

Comment:
Could probably just import the `<Flex />` component and you'll have one less styled component to create.